frontend/src/components/SystemsTable/SystemFilter.tsx

```diff
@@ -0,0 +1,181 @@
+    this.subdataset = partial.subdataset || defaultFilter.subdataset;
+    this.activeSystemIDs =
+      partial.activeSystemIDs || defaultFilter.activeSystemIDs;
+  }
+
+  /*  Update the filter with the values in the new partial filter 
+      Returns true if >= 1 of current filter values get updated
+
+      TODO: Rolling out all if statement seems necessary because Typescript
+      won't allow variable field access for classes. Is there a bette way tho?
+  */
+  update(partial: FilterUpdate): boolean {
+    let updated = false;
+
+    if ((partial.name && this.name !== partial.name) || partial.name === "") {
+      updated = true;
+      this.name = partial.name;
+    }
+
+    if ((partial.task && this.task !== partial.task) || partial.task === "") {
+      updated = true;
+      this.task = partial.task;
+    }
+
+    if (partial.showMine !== undefined && this.showMine !== partial.showMine) {
+      updated = true;
+      this.showMine = partial.showMine;
+    }
+
+    if (partial.sortField && this.sortField !== partial.sortField) {
+      updated = true;
+      this.sortField = partial.sortField;
+    }
+
+    if (partial.sortDir && this.sortDir !== partial.sortDir) {
+      updated = true;
+      this.sortDir = partial.sortDir;
+    }
+
+    if (partial.dataset && this.dataset !== partial.dataset) {
+      updated = true;
+      this.dataset = partial.dataset;
+    }
+
+    if (partial.split && this.split !== partial.split) {
+      updated = true;
+      this.split = partial.split;
+    }
+
+    if (partial.subdataset && this.subdataset !== partial.subdataset) {
+      updated = true;
+      this.subdataset = partial.subdataset;
+    }
+
+    if (
+      partial.activeSystemIDs &&
+      JSON.stringify(this.activeSystemIDs.sort()) !==
+        JSON.stringify(partial.activeSystemIDs.sort())
+    ) {
+      updated = true;
+      this.activeSystemIDs = partial.activeSystemIDs;
+    }
+
+    return updated;
+  }
+
+  toUrlParams(): URLSearchParams {
+    const queryParams = new URLSearchParams();
+    for (const [key, val] of Object.entries(
+      SystemFilter.parseFilterToQuery(this)
+    )) {
+      queryParams.append(key, val);
+    }
+    return queryParams;
+  }
+
+  static getDefaultFilter(): SystemFilter {
+    return new SystemFilter(null);
+  }
+
+  static parseQueryToFilter(query: URLSearchParams): SystemFilter {
+    const defaultFilter = SystemFilter.getDefaultFilter();
+
+    const activeSystemIDs = query.get(filterKeyMap.activeSystemIDs);
+    let activeSystemIDArray: string[];
+    if (activeSystemIDs) {
+      activeSystemIDArray = activeSystemIDs.split(filterDelim);
+    } else {
+      activeSystemIDArray = [];
+    }
+
+    return new SystemFilter({
+      name: query.get(filterKeyMap.nameFilter) || defaultFilter.name,
+      task: query.get(filterKeyMap.taskFilter) || defaultFilter.task,
+      showMine:
+        query.get(filterKeyMap.showMine) === "true"
+          ? true
+          : defaultFilter.showMine,
```

Comment:
default values are applied once here and another time in the constructor. This may introduce inconsistency. I think one approach is to pass in undefined if the filter is not present in the query instead of the actual default value.